package.json

```diff
@@ -143,7 +143,7 @@
     "tslib": "1.6.0",
     "tslint": "^5.0.0",
     "tsutils": "2.20.0",
-    "typescript": "~2.7.2",
+    "typescript": "~2.8.4",
```

Comment:
I think this means that angular version would need a bump as well:
```
@angular/compiler-cli@6.0.9 requires typescript@'>=2.7.0 <2.8.0' but 2.8.4 was found instead.
```

Reply:
Indeed, at this point this PR is doing too much. Another PR is needed to get to TS 2.8, with all the complications involved around angular, bazel, router(?). I can't do that upgrade, but will rebase this PR once that happens.